src/core/lib/channel/http_client_filter.c

```diff
@@ -297,7 +297,7 @@ static grpc_error *hc_mutate_op(grpc_exec_ctx *exec_ctx,
         const int k_url_safe = 1;
         const int k_multi_line = 0;
         const char *k_query_separator = "?";
-        const size_t k_query_separator_len = GPR_ARRAY_SIZE(k_query_separator);
+        const size_t k_query_separator_len = 1; /* strlen(k_query_separator) */
```

Comment:
The original comment has the whole suggestion: https://github.com/grpc/grpc/pull/10222#discussion_r107557882